### PR TITLE
feat(Core/Player): GetFreeInventorySpace()

### PR DESCRIPTION
**Pull request by Kitzunu**
## Changes Proposed:
-  Implement GetFreeInventorySpace() from TrinityCore
- change p_caster to player so this PR is the base for the two other

## Tests Performed:
- Builds

## NEEDED FOR:
- https://github.com/azerothcore/azerothcore-wotlk/pull/4298
- https://github.com/azerothcore/azerothcore-wotlk/pull/4296

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

File: src/server/game/Entities/Player/Player.cpp [hunk 1]
```diff
@@ -10636,6 +10636,28 @@ Bag* Player::GetBagByPos(uint8 bag) const
     return nullptr;
 }
 
+uint32 Player::GetFreeInventorySpace() const
+{
+    uint32 freeSpace = 0;
+
+    // Check backpack
+    for (uint8 slot = INVENTORY_SLOT_ITEM_START; slot < INVENTORY_SLOT_ITEM_END; ++slot)
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.cpp:10644]: could `slot` be `auto` ?

File: src/server/game/Entities/Player/Player.h [hunk 2]
```diff
@@ -1242,6 +1242,7 @@ class Player : public Unit, public GridObject<Player>
     [[nodiscard]] Item* GetItemByPos(uint16 pos) const;
     [[nodiscard]] Item* GetItemByPos(uint8 bag, uint8 slot) const;
     [[nodiscard]] Bag*  GetBagByPos(uint8 slot) const;
+    uint32 GetFreeInventorySpace() const;
```
Comment (FrancescoBorzi) [inline src/server/game/Entities/Player/Player.h:1245]: shouldn't this be ` [[nodiscard]]` as well?

Comment (Kitzunu) [reply src/server/game/Entities/Player/Player.h:1245]: I dont know, I didnt want to put it to wait feedback since I saw some didn't use [[nodiscard]]

Comment (Kitzunu) [reply src/server/game/Entities/Player/Player.h:1245]: gonna add

File: src/server/game/Entities/Player/Player.cpp (hunk 1 above)
Comment (Kitzunu) [reply src/server/game/Entities/Player/Player.cpp:10644]: anything is possible to test :D

**Review by FrancescoBorzi — APPROVED** [review]

**Pull request merged**